app/src/main/java/org/jellyfin/mobile/utils/PermissionRequestHelper.kt

```diff
@@ -0,0 +1,56 @@
+package org.jellyfin.mobile.utils
+
+import android.app.Activity
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import androidx.core.app.ActivityCompat.requestPermissions
+import org.koin.android.ext.android.getKoin
+
+class PermissionRequestHelper {
+    private val permissionRequests = mutableMapOf<Int, (Map<String, Int>) -> Unit>()
+    private var requestCode = 50000 // start at a high number to prevent collisions
+
+    fun getRequestCode() = requestCode++
+
+    fun addCallback(requestCode: Int, callback: (Map<String, Int>) -> Unit) {
+        permissionRequests[requestCode] = callback
+    }
+
+    fun handleRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        // Change to a map
+        val permissionsMap = permissions
+            .mapIndexed { index, permission ->
+                Pair(permission, grantResults.getOrElse(index) { PackageManager.PERMISSION_DENIED })
+            }
+            .toMap()
+
+        // Execute and remove if it exists
+        permissionRequests.remove(requestCode)?.invoke(permissionsMap)
+    }
+}
+
+fun Activity.requestPermission(permission: String, callback: (Map<String, Int>) -> Unit) =
+    requestPermissions(setOf(permission), callback)
+
+fun Activity.requestPermissions(
+    permissions: Collection<String>,
```

Comment:
Wouldn't a vararg array also work?

Reply:
Should work but I'm normally not a big fan of those.